fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionDescBP.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.base;
+
+import com.google.common.base.Preconditions;
+import com.starrocks.connector.BucketProperty;
+
+import java.util.List;
+
+public class HashDistributionDescBP extends HashDistributionDesc {
```

Comment:
add some comment to desc what's this class used for.

Reply:
It's used for hashDistribution but we need bucket property to do further check, for example, left table and right table in a join are both hashDistribution, we need to check their bucket properties to do colocate join or not.